kafka-rest/src/test/java/io/confluent/kafkarest/v2/LoadTest.java

```diff
@@ -0,0 +1,172 @@
+package io.confluent.kafkarest.v2;
+
+import io.confluent.kafkarest.ConsumerReadCallback;
+import io.confluent.kafkarest.KafkaRestConfig;
+import io.confluent.kafkarest.SystemTime;
+import io.confluent.kafkarest.entities.BinaryConsumerRecord;
+import io.confluent.kafkarest.entities.ConsumerInstanceConfig;
+import io.confluent.kafkarest.entities.ConsumerRecord;
+import io.confluent.kafkarest.entities.ConsumerSubscriptionRecord;
+import io.confluent.kafkarest.entities.EmbeddedFormat;
+import io.confluent.rest.exceptions.RestException;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.common.TopicPartition;
+import org.easymock.Capture;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockRunner;
+import org.easymock.IExpectationSetters;
+import org.easymock.Mock;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+
+@RunWith(EasyMockRunner.class)
+public class LoadTest {
```

Comment:
Nice work on this. This seems pretty good for verifying the backoff behavior. I guess the main gap is that we don't have a good way to simulate the interaction with the rebalance protocol. It might also be nice to be able to mix offset commit calls. Anyway, it's a good start.

Reply:
I'm not sure how easy it would be to verify the rebalance protocol. The `MockConsumer` does not seem to respect other consumers that have joined the group - you only schedule the records it should return on `poll()`.
We could modify the callback we give it to mimic the rebalances. I can open a JIRA for that if you think it's a reasonable approach